src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs

```diff
@@ -30,34 +28,32 @@ public static class StandardLearnersCatalog
         /// <param name="maxIterations">The maximum number of iterations; set to 1 to simulate online learning.</param>
         /// <param name="initLearningRate">The initial learning rate used by SGD.</param>
         /// <param name="l2Weight">The L2 regularization constant.</param>
-        /// <param name="loss">The loss function to use.</param>
-        public static StochasticGradientDescentClassificationTrainer StochasticGradientDescent(this BinaryClassificationCatalog.BinaryClassificationTrainers catalog,
+        public static SgdBinaryTrainer StochasticGradientDescent(this BinaryClassificationCatalog.BinaryClassificationTrainers catalog,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
             string weights = null,
-            int maxIterations = SgdOptions.Defaults.MaxIterations,
-            double initLearningRate = SgdOptions.Defaults.InitLearningRate,
-            float l2Weight = SgdOptions.Defaults.L2Weight,
-            ISupportClassificationLossFactory loss = null)
+            int maxIterations = SgdBinaryTrainer.Options.Defaults.MaxIterations,
+            double initLearningRate = SgdBinaryTrainer.Options.Defaults.InitLearningRate,
+            float l2Weight = SgdBinaryTrainer.Options.Defaults.L2Weight)
         {
             Contracts.CheckValue(catalog, nameof(catalog));
             var env = CatalogUtils.GetEnvironment(catalog);
-            return new StochasticGradientDescentClassificationTrainer(env, labelColumn, featureColumn, weights, maxIterations, initLearningRate, l2Weight, loss);
+            return new SgdBinaryTrainer(env, labelColumn, featureColumn, weights, maxIterations, initLearningRate, l2Weight);
```

Comment:
>SgdBinaryTrainer [](start = 23, length = 16)

Why it become `SgdBinaryTrainer` instead of `StochasticGradientDescentClassificationTrainer`?
Method name is  `StochasticGradientDescent`